codegen/src/api/storage.rs

```diff
@@ -61,147 +62,105 @@ fn generate_storage_entry_fns(
     crate_path: &CratePath,
     should_gen_docs: bool,
 ) -> Result<TokenStream2, CodegenError> {
-    let (fields, key_impl) = match storage_entry.entry_type() {
-        StorageEntryType::Plain(_) => (vec![], quote!(vec![])),
+    let keys: Vec<(Ident, TypePath)> = match storage_entry.entry_type() {
+        StorageEntryType::Plain(_) => vec![],
         StorageEntryType::Map { key_ty, .. } => {
             match &type_gen.resolve_type(*key_ty).type_def {
                 // An N-map; return each of the keys separately.
-                TypeDef::Tuple(tuple) => {
-                    let fields = tuple
-                        .fields
-                        .iter()
-                        .enumerate()
-                        .map(|(i, f)| {
-                            let field_name = format_ident!("_{}", syn::Index::from(i));
-                            let field_type = type_gen.resolve_type_path(f.id);
-                            (field_name, field_type)
-                        })
-                        .collect::<Vec<_>>();
-
-                    let keys = fields
-                        .iter()
-                        .map(|(field_name, _)| {
-                            quote!( #crate_path::storage::address::make_static_storage_map_key(#field_name.borrow()) )
-                        });
-                    let key_impl = quote! {
-                        vec![ #( #keys ),* ]
-                    };
-
-                    (fields, key_impl)
-                }
+                TypeDef::Tuple(tuple) => tuple
+                    .fields
+                    .iter()
+                    .enumerate()
+                    .map(|(i, f)| {
+                        let ident: Ident = format_ident!("_{}", syn::Index::from(i));
```

Comment:
Nit; is `Ident` type explicitly needed?

Reply:
Ah, yes it is needed because of the `quote!( #field_name: impl ::std::borrow::Borrow<#field_ty> )` in Line 136, where `field_name` is the `ident` from above.